### ISSUE TITLE
Добавить Callback-уведомления о просмотрах рекламы после вызова VKWebAppShowNativeAds

### ISSUE THREAD

**Issue by nesterov021** [VKCOM/vk-bridge#267]
Думаю, была бы полезная функция Callback-уведомлений о просмотрах рекламы. Например, разработчик в настройках приложения указывает URL, куда будут идти уведомления о новых просмотрах рекламы.

**Comment by ilyapishchulin** [comment]
Не понятно, почему это стоит реализовать на стороне хендлера, а не на стороне приложения?

**Comment by nesterov021** [comment]
На стороне приложения адрес будет виден, легко подделать запрос. т.к проверку на клиенте трудновато сделать. Насчёт существующих, то request_id тоже легко другой передать. Поэтому, думаю, callback-уведомления от VK решат проблему подделки запросов.

**Comment by ilyapishchulin** [comment]
Что будет мешать злоумышленнику отправить запрос на бэкенд вк об успешном просмотре рекламы?

**Comment by eolme** [comment]
@ilyapishchulin, просто ради интереса, в курсе как работает callback, например, боты?

**Comment by ilyapishchulin** [comment]
Конечно, но можно посмотреть как устроена реклама на вебе и всё понять, почему данная реализация не сработает

**Comment by eolme** [comment]
@ilyapishchulin, и видимо все боты тоже не работают, ведь можно посмотреть, что отправляется на севера ВК

**Comment by nesterov021** [comment]
добавить подпись от ВК. Таким образом злоумышленнику всё же будет сложнее подделать запрос

**Comment by nesterov021** [comment]
Можно представить такую реализацию:
Первое уведомление отправляется при открытии рекламы. VK передаёт мне доверенный request_id, я его сохраняю. На стороне VK данный параметр можно проверить, сформирован ли параметр правильно (а не 12340). 
После просмотра отправляется снова мне запрос с тем же requiest_id, и на стороне VK также проверяется request_id. Мне передаётся уже с подписями VK для исключения подделки запроса напрямую ко мне.
Я думаю, что-то можно сделать

**Comment by ilyapishchulin** [comment]
Если посмотреть как реализована реклама в вебе, то можно понять, что она не полностью завязана на сервера ВКонтакте, к сожалению.

Мы постараемся это сделать, но не в ближайшем времени

**Comment by drvilclash** [comment]
Лучше просто добавить sign после просмотра рекламы

**Comment by Pronin-vk** [comment]
@nesterov021  Добрый день, можно передать request_id с клиента и подписаться VKWebAppShowNativeAdsResult и VKWebAppShowNativeAdsFailed, в которых придет request_id, переданный с клиента
